src/ext/gui.rs

```diff
@@ -1,23 +1,64 @@
 use crate::{host::*, plugin::*};
 
-use std::os::raw::c_char;
+use std::ffi::c_void;
+use std::os::raw::{c_char, c_ulong};
 
 pub const CLAP_EXT_GUI: *const c_char = b"clap.gui\0".as_ptr() as *const c_char;
 
+pub const CLAP_WINDOW_API_WIN32: *const c_char = b"win32\0".as_ptr() as *const c_char;
+pub const CLAP_WINDOW_API_COCOA: *const c_char = b"cocoa\0".as_ptr() as *const c_char;
+pub const CLAP_WINDOW_API_X11: *const c_char = b"x11\0".as_ptr() as *const c_char;
+pub const CLAP_WINDOW_API_WAYLAND: *const c_char = b"wayland\0".as_ptr() as *const c_char;
+
+pub type clap_hwnd = *mut c_void;
+pub type clap_nsview = *mut c_void;
+pub type clap_xwnd = c_ulong;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct clap_window {
+    pub api: *const c_char,
+    pub specific: clap_window_handle,
+}
+
+/// Defined as an anonymous union in [`clap_window` ]in the C-version.
```

Comment:
Inconsequential nitpick: fix the space before `]` here?

Reply:
Fixed.